### ISSUE TITLE
Azure Functions Node JS Stack - Context Logging not in Sequential order

### ISSUE THREAD

**Issue by HariKrishna598** [Azure/azure-functions-nodejs-worker#700]
Created the basic Http Trigger in Node Js 18 LTS Azure Functions using VS Code.

```
module.exports = async function (context) {
    context.log('JavaScript HTTP trigger function processed a request.');

    context.log.info({ a: 'first object', b: { q: 'test 1' } });
    context.log.info({ a: 'second object', b: { q: 'test 2' } });
    context.log.info({ a: 'third object', b: { q: 'test 3' } });
  
    context.log.info(JSON.stringify({ a: 'first string', b: { q: 'test 1' } }));
    context.log.info(JSON.stringify({ a: 'second string', b: { q: 'test 2' } }));
    context.log.info(JSON.stringify({ a: 'third string', b: { q: 'test 3' } }));
  
    context.log.info('first simple string');
    context.log.info('second simple string');
    context.log.info('third simple string');
    const responseMessage = "Hello Krishna, This Http Triggered Function executed successfully."

    context.res = {
        // status: 200, /* Defaults to 200 */
        body: responseMessage
    };
}
```

The way I have written my log statements are not coming in the terminal with the same order when I run my function.

![image](https://user-images.githubusercontent.com/88704316/224697100-b4f194c5-6cce-4bb4-8dfe-669e3670df83.png)

Is it the behavior of `Context.Log()` or am I missing something?

**Comment by ejizba** [comment]
@RohitRanjanMS do you know if this is unique to Node.js or does it happen for most languages? I'm wondering if this is just a known limitation of some part of the functions runtime.

**Comment by RohitRanjanMS** [comment]
Most of these are processed asynchronously, so the sequence is not guarantee.

@ejizba , @HariKrishna598

**Comment by ejizba** [comment]
Closing as by-design/upstream per Rohit's response. Attempting to get these in order would likely be more work than it was worth and probably have other negative side effects to things like performance.

**Comment by oshihirii** [comment]
How does anyone log information in their apps?  I tried using Pino, but the logs don't show up in live production environment (ie when calling live function endpoint), only in local development environment (when calling local function endpoint).

**Comment by ejizba** [comment]
Yeah the host team is actively working on this. I can't predict when it'll be fixed, but things are definitely looking a lot better than my original comment in July when I thought this would never happen. I'll reopen this issue to track from a Node.js perspective, but again the work is on the host team which we (the Node.js side of things) don't control

I'm not super familiar with Pino. If you need help with that, please file a new issue with sample code/repro steps as I don't think it's super related to the "out of order" logging
